posydon/visualization/plot_pop.py

```diff
@@ -251,4 +258,137 @@ def plot_hist_properties(x, df_intrinsic=None, df_observable=None,
+            if 'CO-HMS_RLO' in grid_path:
+                sel = (pop.df['metallicity'] == met_Zsun*Zsun) & (pop.df['event'] == 'oRLO2')
+                m_CO = pop.df['S1_mass'].values
+                mask = sel & (m_CO>=slice_3D_var_range[0]) & (m_CO<=slice_3D_var_range[1])
+            elif 'CO-HeMS' in grid_path:
+                sel = (pop.df['metallicity'] == met_Zsun*Zsun) & (pop.df['step_names'] == 'step_CE')
+                m_CO = pop.df['S1_mass'].values
+                mask = sel & (m_CO>=slice_3D_var_range[0]) & (m_CO<=slice_3D_var_range[1])
+            else:
+                raise ValueError('Grid type not supported!')
+        # TODO: skip plotting slice if there are no data
+        try:
+            PLOT_PROPERTIES = {
+                'figsize' : (4,3.5) if prop is None else (4,5.5),
+                'path_to_file' : plot_dir,
+                'show_fig' : False,
+                'close_fig' : False,
+                #'fname' : fname%var,
+                'title' : title%var if channel is None else title%var + '\n' + channel,
+                'log10_x' : True,
+                'log10_y' : True,
+                'legend2D' : {'bbox_to_anchor' : (1.03, 0.5)},
+            }
+            # grid
+            grid.plot2D('star_1_mass', 'period_days', None,
+                            termination_flag='combined_TF12',
+                            grid_3D=True, slice_3D_var_str=slice_3D_var_str,
+                            slice_3D_var_range=slice_3D_var_range,
+                            verbose=False, **PLOT_PROPERTIES)
+            # pop synth
+            # only plot selected channel
+            if channel is not None:
+                sel_binary_index = pop.df_oneline.loc[pop.df_oneline['channel'] == channel].index.values.tolist()
+                mask = mask & (pop.df.index.isin(sel_binary_index))
+            log10_m1 = np.log10(pop.df.loc[mask,'S1_mass'].values)
+            log10_p = np.log10(pop.df.loc[mask,'orbital_period'].values)
+            # plot color map of a given DCO variable
+            if prop is not None:
+                sel = pop.df.index.isin(pop.df.loc[mask].index.values.tolist()) & (pop.df['event'] == 'CO_contact')
+                prop_values = pop.df.loc[sel,prop].values
+                vmin = prop_range[0]
+                vmax = prop_range[1]
+                if log_prop:
+                    prop_values = np.log10(prop_values)
+                    vmin = np.log10(vmin)
+                    vmax = np.log10(vmax)
+                plt.scatter(log10_m1, log10_p, c=prop_values, s=s, vmin=vmin, vmax=vmax, marker='v', cmap='viridis', alpha=alpha)
+                if prop in DEFAULT_LABELS:
+                    if log_prop:
+                        label = DEFAULT_LABELS[prop][1]
+                    else:
+                        label = DEFAULT_LABELS[prop][0]
+                else:
+                    label = prop
+                    if log_prop:
+                        label = 'log10_'+label
```

Comment:
This should read `label = 'log_10('+label+')'`

Reply:
Will not address. Properties without specified default legend are always printed with the variable name string or with `log10_` in front.